### DISCUSSION TITLE
Is there a way to pass route props to navigateTo() ?

### DISCUSSION THREAD

**Discussion by SPH73** [nuxt/nuxt#15969] [Questions] [answered]
More specifically, after successfully submitting a form (and getting back an id) to the database, I'd like to pass the form data and the id and render it in a printable success page.

I'm trying something like this... :

```
//parties/quote.vue
const res = await $fetch("/api/parties/partyQuote", {
    method: "post",
    body: partyData.value,
  });
 
  quoteRef.value = res.id;
  
  await navigateTo({ path: "/success", query: {partyData: partyData.value, quoteRef: quoteRef.value} });
```

I know this way isn't correct but what is the correct way to pass data or props to navigateTo() ?

and then trying to access it in the new route... :
```
<script lang="ts" setup>
defineProps(["partyData", "quoteRef"]);
</script>

<template>
  <div class="max-w-7xl mx-auto">
    <div class="container py-8">
      <BaseButton>Print This Page</BaseButton>
      <h1 class="font-play">Success!</h1>
      <p class="tracking-wide">
        Thank you, we have received your enquiry and will be in contact soon!
      </p>
      <div>
        <h2 class="font-play">Your enquiry details:</h2>
        <p>Quote reference: {{ quoteRef }}</p>
        <p>Your submitted details:</p>
        <pre>{{ partyData }}</pre>
      </div>
    </div>
  </div>
</template>

```

How do I access the data within the success page to display it in the template?

Any help would be super! TIA

**Comment by IsraelOrtuno** [comment, marked as answer]
Not sure what `navigateTo` accepts but will tell you what I do in this case. I use the router to replace the current route since usually you don't want that route to be accessible again via back button navigation:

```js
const router = useRouter()
router.replace({ path: "/success", query: {partyData: partyData.value, quoteRef: quoteRef.value} })
```

And then in the `success` route you can read those query parameters from the route instance:

```js
const route = useRoute()
// route.query.partyData
```

That page itself could have a [Route Validation](https://nuxt.com/docs/getting-started/routing#route-validation) to make sure that those query parameters are in the URL and prevent the access otherwise.

Not sure if a Page Component can accept parameters via routes, that would be interesting to now but since that URL could also be accessed by anywhere else (let's say an email), I think the above solution should work.

I hope this helps!

**Reply by SPH73** [reply]
Thank you, that's worked. 

If I have anything to add that will improve my post I will edit it later but for clarity, this is what I changed it to:

```
//parties/quote.vue
const res = await $fetch("/api/parties/partyQuote", {
    method: "post",
    body: partyData.value,
  });

  quoteRef.value = res.id;
  const router = useRouter();
  router.replace({
    path: "/success",
    query: { partyData: partyData.value, quoteRef: quoteRef.value },
  });
```

```
// success.vue
<script lang="ts" setup>
const route = useRoute();
const partyData = ref(route.query.partyData);
const quoteRef = ref(route.query.quoteRef);
</script>

<template>
  <div class="max-w-7xl mx-auto">
    <div class="container py-8">
      <BaseButton>Print</BaseButton>
      <h1 class="font-play">Success!</h1>
      <p class="tracking-wide">
        Thank you, we have received your enquiry and will be in contact soon!
      </p>
      <div>
        <h2 class="font-play">Your enquiry details:</h2>
        <p>Quote reference: {{ quoteRef }}</p>
        <p>Your submitted details:</p>
        <pre>{{ partyData }}</pre>
      </div>
    </div>
  </div>
</template>
```

Currently, partyData is read as `[Object Object]` and printed to the page like that but `quoteRef` is printing the value. So I just need to figure out how to pass in the data correctly.
